### PR TITLE
Added mocks for Raid and Raider

### PR DESCRIPTION
**Pull request by 4everTheOne**
# Description
Implemented mocks for [Raid](https://jd.papermc.io/paper/1.20.4/org/bukkit/Raid.html) and [Raider](https://jd.papermc.io/paper/1.20.4/org/bukkit/entity/Raider.html). Raider is a abstract class that will be extended by [Evoker](https://jd.papermc.io/paper/1.20.4/org/bukkit/entity/Evoker.html), [Illager](https://jd.papermc.io/paper/1.20.4/org/bukkit/entity/Illager.html), [Illusioner](https://jd.papermc.io/paper/1.20.4/org/bukkit/entity/Illusioner.html), [Pillager](https://jd.papermc.io/paper/1.20.4/org/bukkit/entity/Pillager.html), [Ravager](https://jd.papermc.io/paper/1.20.4/org/bukkit/entity/Ravager.html), [Spellcaster](https://jd.papermc.io/paper/1.20.4/org/bukkit/entity/Spellcaster.html), [Vindicator](https://jd.papermc.io/paper/1.20.4/org/bukkit/entity/Vindicator.html) and [Witch](https://jd.papermc.io/paper/1.20.4/org/bukkit/entity/Witch.html).

# Checklist
The following items should be checked before the pull request can be merged.
- [X] Code follows existing style.
- [X] Unit tests added (if applicable).

# Relates
- https://github.com/MockBukkit/MockBukkit/issues/825

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/RaidMock.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+		this.location = location;
+	}
+
+	@Override
+	public boolean isStarted()
+	{
+		return this.started;
+	}
+
+	public void setStarted(boolean started)
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/RaidMock.java:63]: As this method does not override anything, could you add a description to it?

File: src/main/java/be/seeseemelk/mockbukkit/RaidMock.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+		this.started = started;
+	}
+
+	@Override
+	public long getActiveTicks()
+	{
+		return this.activeTicks;
+	}
+
+	public void setActiveTicks(long activeTicks)
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/RaidMock.java:79]: Description here as well

File: src/main/java/be/seeseemelk/mockbukkit/RaidMock.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+	}
+
+	@NotNull
+	@Override
+	public RaidStatus getStatus()
+	{
+		return this.status;
+	}
+
+	public void setStatus(RaidStatus status)
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/RaidMock.java:116]: And here

File: src/main/java/be/seeseemelk/mockbukkit/RaidMock.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+		return this.groupCount + (getBadOmenLevel() > 1 ? 1 : 0);
+	}
+
+	@Override
+	public int getTotalWaves()
+	{
+		return this.groupCount;
+	}
+
+	public void setGroupCount(int groupCount)
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/RaidMock.java]: Here

File: src/main/java/be/seeseemelk/mockbukkit/RaidMock.java [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+	{
+		Preconditions.checkArgument(groupCount >= 0, "groupCount cannot be negative");
+		this.groupCount = groupCount;
+	}
+
+	public void setGroupCount(Difficulty difficulty)
```
Comment (Thorinwasher) [inline src/main/java/be/seeseemelk/mockbukkit/RaidMock.java]: Here

**Review by Thorinwasher — CHANGES_REQUESTED** [review]
A small nitpick. With description I just mean writing small information about input and output for each method.

**Comment by 4everTheOne** [comment]
Thank you for your review and i agree with it. I'll update this later in the day.

**Review by Thorinwasher — APPROVED** [review]
Looks good to me

**Pull request merged**